apps/api/src/server/applications/subscriptions/subscriptions.controller.js

```diff
@@ -0,0 +1,144 @@
+import * as subscriptionRepository from '../../repositories/subscription.respository.js';
+import { eventClient } from '../../infrastructure/event-client.js';
+import { EventType } from '@pins/event-client';
+import { NSIP_SUBSCRIPTION } from '../../infrastructure/topics.js';
+import { buildSubscriptionPayload } from './subscriptions.js';
+import logger from '../../utils/logger.js';
+import { Prisma } from '@prisma/client';
+
+/**
+ * @type {import('express').RequestHandler}
+ * @throws {Error}
+ * @returns {Promise<void>}
+ */
+export async function getSubscription(request, response) {
+	const { query } = request;
+
+	const caseReference = String(query.caseReference);
+	const emailAddress = String(query.emailAddress);
+
+	try {
+		const res = await subscriptionRepository.findUnique(caseReference, emailAddress);
+
+		if (res === null) {
+			response.status(404).send({ errors: { notFound: 'subscription not found' } });
+		} else {
+			response.send(res);
+		}
+	} catch (/** @type {any} */ e) {
+		/** @type {Object<string, string>} */
+		const errors = {};
+		logger.warn(e, 'unhandled error');
+		errors.unknown = e.message || 'unknown';
+		response.status(400).send({ errors });
+	}
+}
+
+/**
+ * @type {import('express').RequestHandler}
+ * @throws {Error}
+ * @returns {Promise<void>}
+ */
+export async function createSubscription(request, response) {
```

Comment:
There's a decent amount of stuff going on here beyond parsing the request and sending a response - wondering if it might be cleaner to move it all into a separate service function or something (think this is a pattern we're using in other places)

Reply:
Yeah that's fair, see https://github.com/Planning-Inspectorate/back-office/pull/1029/commits/eb007c75ee57bc3b39f5cb86e69e42c9089f39d0